cpp/src/gandiva/llvm_generator.h

```diff
@@ -49,10 +51,48 @@ class GANDIVA_EXPORT LLVMGenerator {
   static Status Make(std::shared_ptr<Configuration> config,
                      std::unique_ptr<LLVMGenerator>* llvm_generator);
 
+  static std::shared_ptr<Cache<BaseCacheKey, std::shared_ptr<llvm::MemoryBuffer>>>
+  GetCache();
+
   /// \brief Build the code for the expression trees for default mode. Each
   /// element in the vector represents an expression tree
   Status Build(const ExpressionVector& exprs, SelectionVector::Mode mode);
 
+  /// \brief Build the code for the expression trees for default mode with a LLVM
+  /// ObjectCache. Each element in the vector represents an expression tree
+  template <class KeyType>
+  Status Build(const ExpressionVector& exprs, SelectionVector::Mode mode,
```

Comment:
Better to add the definition in the cc file.

Reply:
I know, but because this uses template, if it is on the cc file this function gets an undefined reference error for it, as just this function is using template.

Reply:
Ok. It looks like there is code duplication with the other build. Should we remove the other build method if it is not used?

Reply:
Yes, I safely removed the duplicated code.